src/middlewares/one-of.ts

```diff
@@ -62,12 +69,25 @@ export function oneOf(
   const middleware = async (req: InternalRequest, _res: any, next: (err?: any) => void) => {
     const surrogateContext = new ContextBuilder().addItem(dummyItem).build();
 
-    // Run each group of chains in parallel, and within each group, run each chain in parallel too.
```

Comment:
This is a perf loss, but I don't see any other way of implementing this.

Reply:
Latest version still is a perf loss. I've added a summary of changes to the PR description.